### ISSUE TITLE
HTTP/2 framing layer error on /query-stream endpoint using Confluent platform on Docker

### ISSUE THREAD

**Issue by stevecalvert** [confluentinc/ksql#6137]
**Describe the bug**
`curl to http://localhost:8088/query-stream` endpoint gives invalid response over HTTP/2

**To Reproduce**
Steps to reproduce the behavior, include:
1. The version of KSQL.
https://github.com/confluentinc/cp-all-in-one, 5.5.1-post branch running on MacOS on localhost.  Followed steps to setup the USERS, PAGEVIEWS and PAGEVIEWS_FEMALE - all good so far

**Expected behavior**
```
curl -X "POST" "http://localhost:8088/query-stream" --http2 -H "Content-Type: application/json; charset=utf-8" -d $'{ "ksql": "SELECT * FROM PAGEVIEWS_FEMALE EMIT CHANGES;", "streamsProperties": {} }' | jq .
```
should stream updates back to the console
**Actual behaviour**
Although use of the `--http2` option results in an `Upgrade: h2c` being applied to the outgoing HTTP/1.1 request, ksqlDB responds with a 404 indicating this endpoint is not available, which is true for HTTP/1, but I was expecting the connection to be upgraded to HTTP/2

**Additional context**
Repeating the above using `--http2-prior-knowledge` also fails with a `Error in the HTTP2 framing layer` .  Wiresharking localhost shows that the request went out as HTTP/2 as expected but ksqlDB (wrongly?) replied with a TCP packet containing a '426 Upgrade Required', and curl client  just replied with an HTTP/2 GOAWAY which seems reasonable since it was expecting HTTP/2 from the outset
Incidentally, I also see this exact same behaviour if I try to use the https://github.com/confluentinc/ksqldb-graphql package.  I added some comments [to an existing bug report](https://github.com/confluentinc/ksqldb-graphql/issues/11) there since I originally thought this was the source of the problem, but seeing the same behaviour with a curl, I'm beginning to think the problem is with the ksqlDB server?  Any help appreciated - I'm really keen to get that GraphQL client working!

**Comment by vcrfxia** [comment]
Hi @stevecalvert , the HTTP/2 endpoints including `/query-stream` were still under development as of CP 5.5, and were feature-flagged off accordingly: https://github.com/confluentinc/ksql/blob/4b859744943cba228e0abcf1f0a29470fb229d53/ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java#L248
If you really want to enable the new API, you can enable the feature flag, but be warned that the associated feature set was not complete at the time. For example, the HTTP/2 API only supported HTTPS at the time, and not HTTP, so there are additional required configs including https://github.com/confluentinc/ksql/blob/4b859744943cba228e0abcf1f0a29470fb229d53/ksqldb-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java#L54
The HTTP/2 server was also separate from the old HTTP/1.x server, and therefore was hosted on a different port at the time: https://github.com/confluentinc/ksql/blob/4b859744943cba228e0abcf1f0a29470fb229d53/ksqldb-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java#L46
A number of bugs have been fixed since then as well.

All said, I strongly recommend updating to a newer version of ksqlDB if you want to use the HTTP/2 API. In later versions (including in CP 6.0, once released), the HTTP/2 API is supported natively -- no need to explicitly enable it, no separate port, no partial feature set, bunch of bug fixes, etc.

I'm going to close this ticket. Feel free to reopen if I've missed something. Thanks.

**Comment by stevecalvert** [comment]
Many thanks @vcrfxia, very much appreciate your help.  Could you let me know when CP 6.0 will be available as a Docker  standalone package (or point me to where this info is available)?  I tried to find a release later than 5.5.1 to try out, but the only branch that looked hopeful was `6.0.x`.  When I ran  `docker-compose up -d` on this branch, I immediately saw:
`➜  cp-all-in-one git:(6.0.x) docker-compose up -d
Pulling zookeeper (confluentinc/cp-zookeeper:6.0.x-latest)...
ERROR: manifest for confluentinc/cp-zookeeper:6.0.x-latest not found: manifest unknown: manifest unknown`

I guess I need to build the containers manually but I'm not sure how to go about that.  Sorry for my ignorance but what's the easiest way to update to a newer version of ksqlDb?

**Comment by vcrfxia** [comment]
The easiest way to update to a newer version of ksqlDB is to download the Docker distribution of a standalone ksqlDB release: https://ksqldb.io/quickstart.html

The latest released version is ksqlDB 0.11.0. CP 5.5.1 is roughly equivalent to ksqlDB standalone 0.7.1, so there are lots of new features and improvements you'd be getting by updating to 0.11.0 [[changelog](https://github.com/confluentinc/ksql/blob/master/CHANGELOG.md) ; blogs: [1](https://www.confluent.io/blog/ksqldb-0-8-0-feature-updates/), [2](https://www.confluent.io/blog/ksqldb-0-9-0-feature-updates/), [3](https://www.confluent.io/blog/ksqldb-0-10-0-latest-features-updates/), [4](https://www.confluent.io/blog/ksqldb-0-11-0-features-and-improvements/)]

If you insist on using a Confluent Platform version rather than a standalone ksqlDB version, you can wait for Confluent Platform 6.0 which will be released soon (latest [press release](https://www.confluent.io/press-release/project-metamorphosis-global-confluent-platform-6.0/)).

**Comment by stevecalvert** [comment]
Thank you @vcrfxia, moving to ksqlDB 0.11.0 has resolved my issues 😄
